dask/array/core.py

```diff
@@ -1080,8 +1079,12 @@ def store(
 
     if isinstance(sources, Array):
         sources = [sources]
+
+    if isinstance(targets, Array):
```

Comment:
targets can be
- *any* array-like that supports slice-insertion. Specifically, *not* a dask array!
- a dask delayed
- a collection thereof

All sorts of unit tests should go red with this change, and if they don't it should be worrying.